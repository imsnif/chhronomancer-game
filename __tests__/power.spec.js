import 'react-native'
import React from 'react'
import Power from '../components/power'
import renderer from 'react-test-renderer'

jest.mock('mobx-react/native', () => require('mobx-react/custom'))
jest.useFakeTimers()

test('Power (default props) => renders correctly', () => {
  const tree = renderer.create(
    <Power />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Power (with progress) => renders correctly', () => {
  const tree = renderer.create(
    <Power progress={50}/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Power (with timeLeft) => renders correctly', () => {
  const tree = renderer.create(
    <Power timeLeft={'00:00:25'}/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Power (with allied/enemy players) => renders correctly', () => {
  const tree = renderer.create(
    <Power alliedPlayers={1} enemyPlayers={3}/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Power (with progress, timeLeft and allied/enemy players) => renders correctly', () => {
  const tree = renderer.create(
    <Power progress={50} timeLeft={'00:00:25'} alliedPlayers={1} enemyPlayers={3}/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
