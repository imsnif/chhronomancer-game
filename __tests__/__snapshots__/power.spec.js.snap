// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Power (default props) => renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#329932",
      "fontFamily": "telegrama_raw",
      "fontSize": 6,
      "marginLeft": 9,
      "marginTop": 15,
    }
  }
>
  undefined | |--------------------------------------------------| undefined% undefined
</Text>
`;

exports[`Power (with allied/enemy players) => renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#329932",
      "fontFamily": "telegrama_raw",
      "fontSize": 6,
      "marginLeft": 9,
      "marginTop": 15,
    }
  }
>
  undefined | |--------------------------------------------------| undefined% undefined
</Text>
`;

exports[`Power (with progress) => renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#329932",
      "fontFamily": "telegrama_raw",
      "fontSize": 6,
      "marginLeft": 9,
      "marginTop": 15,
    }
  }
>
  undefined | |==========================------------------------|  50% undefined
</Text>
`;

exports[`Power (with progress, timeLeft and allied/enemy players) => renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#329932",
      "fontFamily": "telegrama_raw",
      "fontSize": 6,
      "marginLeft": 9,
      "marginTop": 15,
    }
  }
>
  undefined | |==========================------------------------|  50% 00:00:25
</Text>
`;

exports[`Power (with timeLeft) => renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#329932",
      "fontFamily": "telegrama_raw",
      "fontSize": 6,
      "marginLeft": 9,
      "marginTop": 15,
    }
  }
>
  undefined | |--------------------------------------------------| undefined% 00:00:25
</Text>
`;
